packages/data-point/lib/entity-types/base-entity/resolve.js

```diff
@@ -76,39 +67,28 @@ module.exports.createCurrentAccumulator = createCurrentAccumulator
  */
 function resolveMiddleware (manager, name, acc) {
   return middleware.resolve(manager, name, acc).then(middlewareResult => {
-    const reqCtx = utils.assign(acc, {
-      value: middlewareResult.value,
-      locals: middlewareResult.locals
-    })
-
     if (middlewareResult.___resolve === true) {
       // doing this until proven wrong :)
       const err = new Error('bypassing middleware')
       err.name = 'bypass'
       err.bypass = true
-      err.bypassValue = reqCtx
+      err.bypassValue = middlewareResult.value
       return Promise.reject(err)
     }
 
-    return reqCtx
+    return middlewareResult.value
   })
 }
 
 module.exports.resolveMiddleware = resolveMiddleware
 
-function typeCheck (manager, acc, reducer, resolveReducer) {
-  // if no error returns original accumulator
-  // this prevents typeCheckTransform from mutating the value
-  return resolveReducer(manager, acc, reducer).return(acc)
-}
```

Comment:
instead of using this function, I started using `Promise.tap` (since that just returns the input that's given)